vips/VIP-ExtensionV3.md

```diff
@@ -0,0 +1,137 @@
+---
+Title: EVM Clause Context
+Description: This VIP proposes an enhancement of the built-in extension contract to provide information about the clauses in the current transaction.
+Author: @databeforedishonor and Darren Kelly (darren.kelly@vechain.org)
+Discussions: https://vechain.discourse.group/t/mtt-induced-vulnerabilities-in-randomized-events/88
+Category:  Core
+Status: Draft
+CreatedAt: 2024-04-18
+
+---
+
+## Overview
+
+The below VIP-XXX proposal outlines an upgrade to the EVM, which will expose clause information to the smart contract, via the built-in `extension` contract
+  
+## Motivation
+
+Smart contract developers may want to access information about the current clause being executed. For example, a smart contract can limit the number of clauses in case any malicious actor could exploit the contract via multiple clauses.
+
+## Rationale
+
+The design and decision to expose the clause context in the `extension` contract follows the same pattern as [VIP-191](https://github.com/vechain/VIPs/blob/master/vips/VIP-191.md). This time, the `extension` contract will migrate from V2 to V3 to provide the clause context information.
```

Comment:
Does VIP-191 mentioned the new method in extension contract?

Reply:
It doesn't specify it no, but I guess this part was introduced as part of the VIP:

```solidity
function txGasPayer() public view returns(address) {
        return ExtensionV2Native(this).native_txGasPayer();
}
```

I can remove the specifics if we don't want to be explicit?

Reply:
I would suggest not mentioning VIP-191 since it doesn't include that inside the VIP to avoid any confusion